src/vertex_id.rs

```diff
@@ -1,14 +1,26 @@
 // Copyright 2019 Octavian Oncescu
 
+use rand;
 use rand::Rng;
+use rand::SeedableRng;
+use rand_isaac::IsaacRng;
 
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 /// Id of a vertex
 pub struct VertexId([u8; 16]); // 128bit
 
 impl VertexId {
     pub fn random() -> VertexId {
-        let bytes = rand::thread_rng().gen::<[u8; 16]>();
+        /*
+        use rand;
+        use rand::Rng;
+        use rand_hc::Hc128Rng;
+        use rand::SeedableRng;
+
+        let bytes = Hc128Rng::gen::<[u8; 16]>(&mut Hc128Rng::seed_from_u64(6);
+        */
+        let bytes = IsaacRng::gen::<[u8; 16]>(&mut IsaacRng::seed_from_u64(9));
```

Comment:
The tests probably fail because of `VertexId` collisions. You seem to be feeding a static seed to the rng via the `IsaacRng::seed_from_u64` method. Maybe you could try `IsaacRng::next_u64`?

Reply:
Or we can simply use [this](https://rust-random.github.io/rand/rand_core/trait.RngCore.html#tymethod.fill_bytes).

Reply:
The fill bytes method will still require a seed `IsaacRng::fill_bytes(&mut IsaacRng::seed_from_u64(4), &mut bytes)`

Reply:
@sn99 Well, the problem still stands: the seed is static. Does it work in combination with this: `IsaacRng::next_u64`?

Reply:
Not really but `IssacRng::next_u64` does give more seeding options